server/server.go

```diff
@@ -52,9 +71,15 @@ func ListenAndServe(ctx context.Context, httpsPort, httpPort int, handler http.H
 	if opts.DisplayServerLogs {
 		writer = logger.WriterLevel(logrus.ErrorLevel)
 	}
-	// Otherwise preserve legacy behaviour of displaying server logs only in debug mode.
 
-	errorLog := log.New(writer, "", log.LstdFlags)
+	var errorLog *log.Logger
+	if opts.IgnoreTLSHandshakeError {
+		debugWriter := &TLSErrorDebugger{}
+		errorLog = log.New(debugWriter, "", log.LstdFlags)
+	} else {
+		// Otherwise preserve legacy behaviour of displaying server logs only in debug mode.
+		errorLog = log.New(writer, "", log.LstdFlags)
```

Comment:
Nit: standard flags ends up creating duplicate timestamps, since standard logger adds a timestamp to the payload.

```golang
	l := log.New(&TLSErrorDebugger{}, "", log.LstdFlags)
	l.Print("test")
	l.Print("http: TLS handshake error: EOF")
```

```
time="2009-11-10T23:00:00Z" level=error msg="2009/11/10 23:00:00 test"
time="2009-11-10T23:00:00Z" level=error msg="2009/11/10 23:00:00 http: TLS handshake error: EOF"
```

Just use no flags:

```suggestion
		errorLog = log.New(&TLSErrorDebugger{}, "", 0)
	} else {
		// Otherwise preserve legacy behaviour of displaying server logs only in debug mode.
		errorLog = log.New(writer, "", 0)
```

With this change I think you could also do bytes.HasPrefix instead of bytes.Contains, which is arguably more efficient - but I guess that would only work if the caller is consistent about not adding timestamp flags.